docs/tutorials/new-feature-tutorial.md

```diff
@@ -660,15 +678,36 @@ behavior of the setting you just created.
 
 ### Update documentation
 
-After you add a new view, you should document your feature. This
-feature adds new functionality that requires messages to have topics
-if the setting is enabled. A recommended way to document this feature
-would be to update and/or augment [Zulip's user
-documentation](https://zulip.com/help/) to reflect your changes and
-additions.
-
-At the very least, this will involve adding (or modifying) a Markdown file
-documenting the feature to `templates/zerver/help/` in the main Zulip
-server repository, where the source for Zulip's user documentation is
-stored. For information on writing user documentation, see
+Nice job! You've added a new feature to Zulip that will improve user
+and contributor experiences with the app, which is why it's really
+important to make sure that your new feature is well documented.
+
+This example feature adds new functionality that requires messages to
+have topics if the setting is enabled. A recommended way to document
+this feature would be to update and/or augment
+[Zulip's user documentation](https://zulip.com/help/) to reflect your
+changes and additions.
+
+At the very least, this will involve adding (or modifying) a Markdown
+file documenting the feature to `templates/zerver/help/` in the main
+Zulip server repository, where the source for Zulip's user
+documentation is stored. Details about writing, editing and testing
+these Markdown files are in:
 [Zulip's general user guide documentation](../documentation/user.md).
+
+Also, new features will often impact Zulip's REST API documentation,
+which is found in `zserver/openapi/zulip.yaml`. You may have noticed
+this during the testing process as the Zulip test suite should fail if
+there is a change to the API without a corresponding update to the
+documentation.
+
+The best way to understand writing and updating Zulip's API
+documentation is to read more about Zulip's
+[REST API documentation process](../documentation/api.md)
+and [OpenAPI configuration](../documentation/openapi.md).
+
+Even if you do not need to go through the full API endpoint
+documentation process, **make sure** that you document your new feature
+in `templates/zserver/api/changelog.md` and bump the `API_FEATURE_LEVEL`
+in `version.py` so that there is a clear record of the new feature and
+when it was implemented in the app.
```

Comment:
I'd add a sentence of the form: "The API feature level allows the developers of mobile clients and other tools using the Zulip API to programmatically determine whether the Zulip server they are interacting with has a given feature; see the [Zulip release lifecycle](https://zulip.readthedocs.io/en/latest/overview/release-lifecycle.html) for backend on why this is important."

Reply:
Related but possibly not for this PR, your explanation here helped me understand what this second sentence in the change log is saying ...

> The recommended way for a client like the Zulip mobile or desktop apps that needs to support interaction with a wide range of different Zulip server versions is to check the `zulip_feature_level` parameter in the `/register` and `/server_settings` responses to determine which of the below features are supported.

There are so many clauses in that sentence that even I, a native English speaker, get lost. I think a tiny edit (in bold) would help clarify ...

> The recommended way for **a client, such as the Zulip mobile or desktop apps, that needs** to support interaction with a wide range of different Zulip server versions is to check the `zulip_feature_level` parameter in the `/register` and `/server_settings` responses to determine which of the below features are supported.

Reply:
This is a good thought; I ended up doing a bit more of a rewrite of the heading section of that page: 3871c99104165b6d2fcf417177b3ccd1d3a915f5.

It'd be worth starting a conversation in the "api design" stream about whether there's more we want to do to improve it.